subprojects/gradle/build-trace/src/main/kotlin/com/avito/android/build_trace/internal/TaskExt.kt

```diff
@@ -0,0 +1,29 @@
+package com.avito.android.build_trace.internal
+
+import org.gradle.api.Task
+import org.gradle.api.internal.TaskInternal
+import org.gradle.api.tasks.TaskDependency
+
+internal val Task.type: Class<out Task>
+    get() = (this as TaskInternal).taskIdentity.type
+
+internal val Task.predecessors: Set<Task>
+    get() {
+        val dependenciesByInputs = project.gradle.taskGraph.getDependencies(this)
+        return (
+            dependenciesByInputs +
+                taskDependencies.safeDependencies(this) +
+                mustRunAfter.safeDependencies(this) +
+                shouldRunAfter.safeDependencies(this)
+            )
+            .toSet()
+    }
+
+internal fun TaskDependency.safeDependencies(task: Task): Set<Task> {
+    return try {
+        this.getDependencies(task)
+    } catch (error: Exception) {
```

Comment:
Could we don't swallow this exception and use it to describe explicit result type
I prefer to use Failure then emptySet

Reply:
Ok, wrap it. But clients anyway will skip it.

Reply:
Clients should log can't build critical path because of `exception` and skip the critical path phase

Reply:
We faced a bug in resolving dependencies: https://issuetracker.google.com/issues/142268503
I will change the logic and skip only this specific case. Otherwise, we will lose all the data. This bug was not specific for a critical path but for resolving dependencies in general.

Reply:
Removed the check at all because couldn't reproduce the issue.
https://github.com/avito-tech/avito-android/pull/985/commits/649bc7548b8c45d8cfbd158643d2681249186ab6